src/http_tracker_connection.cpp

```diff
@@ -83,7 +83,7 @@ namespace libtorrent
 	{
 		std::string url = tracker_req().url;
 
-		if (tracker_req().kind == tracker_request::scrape_request)
```

Comment:
wouldn't a more reliable way (i.e. less likely to introduce errors) to make the new i2p flag not be a flag, but just a part of the enumeration, and call it i2p_announce or something like that?